src/azure-cli/azure/cli/command_modules/appconfig/_client_factory.py

```diff
@@ -10,16 +10,29 @@
 logger = get_logger(__name__)
 
 
+STABLE_API_VERSION = '2022-05-01'
+REPLICATION_API_VERSION = '2022-03-01-preview'
+
+
 def get_appconfig_service_client(cli_ctx, api_version=None):
     ''' Returns the client for managing configuration stores.'''
     from azure.mgmt.appconfiguration import AppConfigurationManagementClient
-    return get_mgmt_service_client(cli_ctx, AppConfigurationManagementClient, api_version=api_version)
+    api_version = STABLE_API_VERSION if not api_version else api_version
+    client = get_mgmt_service_client(cli_ctx, AppConfigurationManagementClient, api_version=api_version)
+    client.api_version = api_version
```

Comment:
By defining API version for each operation in cli core, you don't need to specify `api_version` here. `get_mgmt_service_client` will smartly find the exact version.

Reply:
Removed the unnecessary input argument.